PayBySquare.TextGenerator.NET/Code/PayBySquareOverkill.vb

```
@@ -27,7 +27,9 @@ Public Class PayBySquareOverkill
             TS.Append(P.VariableSymbol)
             TS.Append(P.ConstantSymbol)
             TS.Append(P.SpecificSymbol)
-            TS.Append(CStr(Nothing))        'OriginatorsReferenceInformation (VS, SS a KS in SEPA format) - only if VS, KS and SS are empty
+            TS.Append(If(String.IsNullOrWhiteSpace(P.VariableSymbol) AndAlso
```

Let's add validation that will throw InvalidOperationException when PayerReference is not empty, and any of VS, SS or CS is also not empty.

That will make it easier to troubleshoot the question of "why is the PayerReference missing in the output?"

Here, we can then append directly, without an IF

---

Good idea, i added validation.